lib/mayaUsd/fileio/chaser/importChaserRegistry.h

```diff
@@ -0,0 +1,130 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef IMPORTCHASERREGISTRY_H
+#define IMPORTCHASERREGISTRY_H
+
+#include "importChaser.h"
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/fileio/jobs/jobArgs.h>
+
+#include <pxr/base/tf/declarePtrs.h>
+#include <pxr/base/tf/weakBase.h>
+#include <pxr/pxr.h>
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/common.h>
+#include <pxr/usd/usd/primFlags.h>
+
+#include <maya/MDagPathArray.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+TF_DECLARE_WEAK_PTRS(UsdMayaImportChaserRegistry);
+
+/// \class UsdMayaImportChaserRegistry
+/// \brief Registry for import chaser plugins.
+///
+/// We allow sites to register new chaser scripts that can be enabled on post-import.
+///
+/// Use USDMAYA_DEFINE_IMPORT_CHASER_FACTORY(name, ctx) to register a new chaser.
+class UsdMayaImportChaserRegistry : public TfWeakBase
+{
+public:
+    /// \brief Holds data that can be accessed when constructing a
+    /// \p UsdMayaImportChaser object.
+    ///
+    /// This class allows plugin code to only know about the context object
+    /// during construction. All the data it needs during construction should be passed in here.
+    class FactoryContext
+    {
+    public:
+        MAYAUSD_CORE_PUBLIC
+        FactoryContext(
+            Usd_PrimFlagsPredicate&     returnPredicate,
+            const UsdStagePtr&          stage,
+            const MDagPathArray&        dagPaths,
+            const SdfPathVector&        sdfPaths,
+            const UsdMayaJobImportArgs& jobArgs);
+
+        /// \brief Returns a pointer to the imported stage object.
+        MAYAUSD_CORE_PUBLIC
+        UsdStagePtr GetStage() const;
+
+        /// \brief Returns the top-level imported DAG paths that were imported.
+        MAYAUSD_CORE_PUBLIC
+        const MDagPathArray& GetImportedDagPaths() const;
+
+        /// \brief Returns the top-level prims that were imported into the Maya scene.
+        MAYAUSD_CORE_PUBLIC
+        const SdfPathVector& GetImportedPrims() const;
+
+        /// \brief Returns the arguments used for the import operation.
+        MAYAUSD_CORE_PUBLIC
+        const UsdMayaJobImportArgs& GetImportJobArgs() const;
+
+    private:
+        UsdStagePtr                 _stage;
+        const MDagPathArray&        _dagPaths;
+        const SdfPathVector&        _sdfPaths;
+        const UsdMayaJobImportArgs& _jobArgs;
+    };
+
+    typedef std::function<UsdMayaImportChaser*(const FactoryContext&)> FactoryFn;
```

Comment:
We've been favoring `using` over `typedef` in new code.

Reply:
This is addressed in e8efa0c.